extensions/scarb-cairo-run/src/main.rs

```diff
@@ -0,0 +1,106 @@
+use std::collections::HashMap;
+use std::fs;
+
+use anyhow::Result;
+use cairo_lang_runner::short_string::as_cairo_short_string;
+use cairo_lang_runner::{SierraCasmRunner, StarknetState};
+use cairo_lang_sierra::extensions::gas::{
+    BuiltinCostWithdrawGasLibfunc, RedepositGasLibfunc, WithdrawGasLibfunc,
+};
+use cairo_lang_sierra::extensions::NamedLibfunc;
+
+use clap::Parser;
+
+use scarb_metadata::packages_filter::PackagesFilter;
+use scarb_metadata::MetadataCommand;
+
+#[derive(Parser, Clone, Debug)]
+#[command(author, version)]
+struct Args {
+    #[command(flatten)]
+    packages_filter: PackagesFilter,
+    #[arg(long)]
+    available_gas: Option<usize>,
+    #[arg(long, default_value_t = false)]
+    print_full_memory: bool,
+}
+
+fn main() -> Result<()> {
+    let args: Args = Args::parse();
+    let metadata = MetadataCommand::new().inherit_stderr().exec()?;
+
+    let package = args.packages_filter.match_one(&metadata)?;
+    println!("running {} ...", package.name);
+
+    let path = format!(
+        "{}/target/dev/{}.sierra",
+        package.root.into_string(),
+        package.name
+    );
```

Comment:
Please make use of `SCARB_TARGET_DIR` environment variable here instead of assuming things. https://docs.swmansion.com/scarb/docs/extensions/subcommands